tests/test-msgmod.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# This test exercises various base message modules by writing them out to .json
+# files and comparing the number of entries.
+
+script_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+nmsgtool_test=$script_dir/../src/nmsgtool
+indir=$script_dir/private-tests/
+outdir=/tmp/test-msgmod/
+retval=0
+
+check() {
+	if [ $? = "0" ]; then
+		echo "PASS: $*"
+	else
+		echo "FAIL: $*"
+		retval=1
+	fi
+}
+
+# Param. 1 is message type, param. 2 is input .pcap file name, and param. 3 is the expected final entry count.
+run_nmsgtools() {
+	nmsgout=test-$1.nmsg
+	jsonout=test-$1.json
+
+	$nmsgtool_test -ddddd -p $indir/$2 -V base -T $1 -w $outdir/$nmsgout
+	$nmsgtool_test -ddddd -r $outdir/$nmsgout -J $outdir/$jsonout
+
+	line_count=$(wc -l $outdir/$jsonout | awk '{print $1}')
+	final_entry_comparison=0; [ "$line_count" == "$3" ] && final_entry_comparison=1
+	check "base/$1 pcap message parsing ($line_count/$3 payloads parsed)"
+}
+
+echo "Testing message modules: "
+
+mkdir $outdir
+
+# Test message modules (setting a few environment variables to increase coverage).
+export DNSQR_CAPTURE_RD=1
```

Comment:
Exercising functionality but not testing it will just contribute to the illusion that certain code paths are properly tested when in fact they are not.

Reply:
Fixed